//declaring a function as async automatically returns a promise
// async function hello() {

// }

//arrow function syntax for declaring async functions
// const sing = async () => {
//     // throw new Error("UH OH");
//     throw "OH NO PROBLEM";
//     return 'LALALALA';
// }

// sing()
//     .then(data => {
//     console.log('Promise resolved with:', data)
//     })
//     .catch(err => {
//         console.log('OH NO, PROMISE REJECTED')
//         console.log(err)
//     })